extensions/smallrye-jwt/deployment/src/main/java/io/quarkus/smallrye/jwt/deployment/SmallRyeJwtProcessor.java

```diff
@@ -226,4 +245,95 @@ public boolean getAsBoolean() {
             return config.enabled();
         }
     }
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    void generateSignKeys(LiveReloadBuildItem liveReloadBuildItem,
+            BuildProducer<DevServicesResultBuildItem> devServices) {
+
+        if (!config.generateSignKeys()) {
+            return;
+        }
+
+        Config configProvider = ConfigProvider.getConfig();
+        List<String> configuredKeyProperties = JWT_KEY_PROPERTIES.stream()
+                .map(prop -> configProvider.getOptionalValue(prop, String.class).orElse(""))
+                .filter(Predicate.not(String::isBlank))
+                .toList();
+
+        if (!configuredKeyProperties.isEmpty()) {
+            return;
+        }
+
+        ExistingSignKeys existingSignKeys = liveReloadBuildItem.getContextObject(ExistingSignKeys.class);
+        if (existingSignKeys == null && !liveReloadBuildItem.isLiveReload()) {
+            // first execution
+            KeyPair keyPair = getKeyPair();
+            String publicKey = getStringKey(keyPair.getPublic());
+            String privateKey = getStringKey(keyPair.getPrivate());
+
+            ExistingSignKeys signKeys = new ExistingSignKeys();
+            signKeys.addPublicKey(publicKey);
+            signKeys.addPrivateKey(privateKey);
+
+            Map<String, String> properties = generateDevServiceProperties(publicKey, privateKey);
+
+            devServices.produce(new DevServicesResultBuildItem(
+                    Feature.SMALLRYE_JWT.name(), null, properties));
+
+            liveReloadBuildItem.setContextObject(ExistingSignKeys.class, signKeys);
+
+        } else if (existingSignKeys != null && liveReloadBuildItem.isLiveReload()) {
+
+            String publicSignKey = existingSignKeys.getPublicKey();
+            String privateSignKey = existingSignKeys.getPrivateKey();
+
+            Map<String, String> properties = generateDevServiceProperties(publicSignKey, privateSignKey);
+
+            devServices.produce(new DevServicesResultBuildItem(
+                    Feature.SMALLRYE_JWT.name(), null, properties));
+        }
+    }
+
+    private static Map<String, String> generateDevServiceProperties(String publicKey, String privateKey) {
+        return Map.of(
+                MP_JWT_VERIFY_PUBLIC_KEY, publicKey,
+                SMALLRYE_JWT_SIGN_KEY, privateKey);
+    }
+
+    private static String getStringKey(Key key) {
+        return Base64.getEncoder()
+                .encodeToString(key.getEncoded());
+    }
+
+    private KeyPair getKeyPair() {
```

Comment:
There is KeyUtils in the smallrye-jwt project which should get it in a one line call